### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by jebob**
Mostly cleanups and a few fixes. The biggest structural change is replacing the new-format team (a list of seven items) with a class.

Behavioural changes:
* Referee score now includes every round, not just the last round.
* Now search every group for alternative referees, instead of accepting the first group with a valid referee. This lowers the average score (over 5000 sims) from 28.14 to 27.16
* Refreshing now always extends the list of eligible players instead of replacing it. This helps preserve the desired play order

### CODE REVIEW

File: vollyball_scheduling.py [hunk 1]
```diff
@@ -4,11 +4,18 @@
...
 rows repeat: "team1, team2, referee" for each match.
 
-The code in this file was written jointly by Micheal Howlett and David Howlett.
+Tries to minimize score =
+    Number of games where the referee is from a different group * 0.1
+    + (max(games played per team) - min(games played per team)) * 5
+    + sum(max(games played per team) - min(games played per team) for each group) * 2
+    + sum(max(games played or refereed per team) - min(games played or refereed per team) for each group)
+
+The code in this file was written jointly by Michael Howlett and David Howlett.
```
Comment (DavidHowlett) [inline vollyball_scheduling.py]: Please add yourself to the list of authors

File: vollyball_scheduling.py [hunk 2]
```diff
@@ -21,10 +28,12 @@
     ["Team3-01", "Team3-02", "Team3-03", "Team3-04", "Team3-05"],
     ["Team4-01", "Team4-02", "Team4-03", "Team4-04", "Team4-05", "Team4-06"],
 ]
+ALL_TEAMS = [team for group in ORIGINAL_GROUPS for team in group]
+assert len(ALL_TEAMS) == len(set(ALL_TEAMS)), "Duplicate team, or team in >1 group!"
```
Comment (DavidHowlett) [inline vollyball_scheduling.py:35]: Good check

**Comment by DavidHowlett** [comment]
I will look at this properly after work

File: vollyball_scheduling.py (hunk 1 above)
Comment (jebob) [reply vollyball_scheduling.py]: Done

**Comment by DavidHowlett** [comment]
Thanks for your changes, they look good to me.

The tests in this repo are in the pre-commit hook. Currently flake8 and mypy are failing on your branch. Please pass the pre-commit hook checks. I haven't hooked it up to the gitlab CI yet so there is no red CI to show you yet. You can install pre-commit from here: https://pre-commit.com/ or you can run the commands in the hook manually.

I was not expecting any external contributors otherwise I would have put up a proper CI.

**Review by DavidHowlett — APPROVED** [review]
I am happy for this to be merged once the tests in the pre-commit hook pass.

**Comment by DavidHowlett** [comment]
Probably not important but I hit a github bug while doing the review: https://support.github.com/ticket/personal/0/1527208

**Comment by jebob** [comment]
> pre-commit

Now passing.

You might want to update your pre-commit hooks. `pre-commit autoupdate` updates. Black is currently set to "stable", but this is a movable reference so you can get inconsistency issues where one person is on an old "stable" and another is on a new "stable". This is fixed with `pre-commit autoupdate`.

**Comment by DavidHowlett** [comment]
I agree that there is a potential issue with floating versions for things like black. We hit that one at work.

**Comment by DavidHowlett** [comment]
Thanks for your changes

File: vollyball_scheduling.py [hunk 3]
```diff
@@ -248,27 +248,13 @@ def get_score(groups, match_up_score):
     given a partially calculated score, and the datastructure describing the solution,
     calculate the final score
     """
-    min_games = 9999
-    max_games = 0
+    min_games = min(team.games_played for group in groups for team in group)
```
Comment (DavidHowlett) [inline vollyball_scheduling.py:251]: @macfor this is how to write the code you wanted to write for the min and max

**Pull request merged**